### PR TITLE
feat(BAC-52): added endpoint to view all files uploaded in DM

### PR DESCRIPTION
**Pull request by NikkyXO**

### CODE REVIEW

**Review by mikengr — CHANGES_REQUESTED** [review]
Your endpoints weren't documented and is anti-pattern.
Besides that, how did you test these endpoints? @NikkyXO 
Attach screenshots if possible

**Comment by NikkyXO** [comment]
> these
Screenshot no_file1
NoUploadedFile2

NoUploadedFile3

I have added documentation on how i wrote the end point for viewing an files in a room .However Adequate testing couldnt be achieved, especially the one involving getting input from the endpoint handling uploading of  files
The other colleague  which this task was assigned  couldnt finish up with his task, largely due to the fact that we were assigned our various tasks in less than 24 hours.. This was an error made by the product manager who gave us wrong repo with outdated codebase to study and refactor the implemenation of all endpoint relating to file upload in a room as we were mainly freelancers from other group and given the right repo in less than 24 hours when the mistake was realised

File: backend/endpoints/file.py [hunk 1]
```diff
@@ -24,35 +29,61 @@
...
+An Endpoint that handles viwing image files uploaded in the DM
+
+params:
+    org_id = Organisation id of user
+    room_id = room id of user
+
+Retrieves All Messages in the Dm with the get_room_message method
+Handles  conditons for single and Multiple Uploaded Files
+"""
 @router.get("/get/files/{org_id}/{room_id}")
```
Comment (mikengr) [inline backend/endpoints/file.py]: this url path is wrongly defined, check the codebase and observe the url naming patterns

File: backend/endpoints/file.py [hunk 2]
```diff
@@ -24,35 +29,61 @@
...
     data = io.BytesIO()
     Zipfile = zipfile.ZipFile(data, "w")
 
     for fpath in filenames:
-        try:
-            fdir, fname = os.path.split(fpath)
-        except:
-            fpath = fname
+        fdir, fname = os.path.split(fpath)
+        print(fname,fdir)
```
Comment (mikengr) [inline backend/endpoints/file.py]: don't push print statements

File: backend/endpoints/file.py [hunk 3]
```diff
@@ -24,35 +29,61 @@
...
         Zipfile.write(fpath, fname)
     Zipfile.close()
     resp = Response(data.getvalue(), media_type="image/png", headers={
```
Comment (mikengr) [inline backend/endpoints/file.py]: name your variables properly with meaningful names

**Review by mikengr — CHANGES_REQUESTED** [review]
@NikkyXO your changes have been observed, but your screenshots don't show that this endpoint will do what you say it does. Please kindly show screenshots of the different responses gotten when this endpoint is called

File: backend/endpoints/file.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+An Endpoint that handles viewing image files uploaded in the DM
+
+params:
+    org_id = Organisation id of user
+    room_id = room id of user
+
+Retrieves All Messages in the Dm with the get_room_message method
+Handles  conditons for single and Multiple Uploaded Files
+"""
+@router.get("/get/files/{org_id}/{room_id}")
```
Comment (FlorenceEgwu) [inline backend/endpoints/file.py]:
```suggestion
@router.get("/org/{org_id}/rooms/{room_id}/files")
```
your endpoint should look like this instead

File: backend/endpoints/file.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    if  messages:
+        list_files = []
+        for message in messages:
+            if message['files']:
+                list_files.append(message['files'])
+    
+
+        if list_files:
+            if len(list_files):
+                str(list_files)
```
Comment (FlorenceEgwu) [inline backend/endpoints/file.py]: what is the difference between line 96 and 97?

File: backend/endpoints/file.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+            elif len(list_files) > 1:
+                list_files = [''.join(el) for el in list_files]
+                return zipfiles(list_files)
+        else:
+            raise HTTPException(
+                        status_code=status.HTTP_404_NOT_FOUND,
+                        detail="No file available in DM",
+                    )
+
```
Comment (FlorenceEgwu) [inline backend/endpoints/file.py]: Please no 3 level nesting for readability sake.
You could use negative conditionals instead. example:

```suggestion
  if not messages:
       raise Exception
  list_files = []
  for message in messages:
      code block
 if not list_files:
     raise Exception
 list_files ...

 
            
  
```

File: backend/main.py [hunk 7]
```diff
@@ -1,9 +1,10 @@
 from config.settings import settings
-from endpoints import members, messages, rooms, sync, threads
+from endpoints import members, messages, rooms, sync, threads, file
```
Comment (FlorenceEgwu) [inline backend/main.py]: the file name should always be in plural to communicate processing of multiple resource.
so rename `file.py` to `files.py`

**Review by FlorenceEgwu — CHANGES_REQUESTED** [review]

**Comment by FlorenceEgwu** [comment]
Good job!
I have concerns your changes might not work as expected.
kindly record a screen recording of your testing process

**Comment by NikkyXO** [comment]
good morning sir,
 i have made required changes to the end point  and here is  the required
screen shot  of the endpoint testings

The screenshots are as follows
A screenshot  was done while getting the message list and retrieving empty
files no files where uploaded then
because Uploading through the Upload file endpoint was unavailable as the
task is unfinished by another, and the send messages endpoint
kep giving 422 error, i used a dummy urlpath for testing the endpoint and
helper zipfile file function
screenshot was done while trying to retrieve and view archive files

the screenshots are attached below, thanks for your kind patience

yours sincerely
olanike ogundiran
[image: CheckingArchiveFile.png]
[image: dummyurlpath.png][image: extract3.png][image:
extractArchive.png][image:
extractedArchive.png][image: list_of_files.png][image:
List_of_uploaded_files.png]

**Comment by NikkyXO** [comment]
here is the link to the Pulll Request:
https://github.com/zurichat/zc_messaging/pull/146

**Comment by NikkyXO** [comment]
i unintentional made a git push -force from my branch while pushing to
github , this closed the previous pull request i made
here is the new pull request:
https://github.com/zurichat/zc_messaging/pull/155

Apologies for the inconvenience
yours sincerely
olanike

**Pull request closed without merging**